services/src/main/java/org/exoplatform/addons/gamification/service/AnnouncementService.java

```diff
@@ -31,7 +32,7 @@ List<Announcement> findAllAnnouncementByChallenge(long challengeId, int offset,
    * @throws IllegalAccessException when user is not authorized to create a
    *           announcement for the designated owner defined in object
    */
-  Announcement createAnnouncement(Announcement announcement, String username, boolean system) throws IllegalAccessException,
+  Announcement createAnnouncement(Announcement announcement, Map<String, String> templateParams , String username, boolean system) throws IllegalAccessException,
```

Comment:
```suggestion
  Announcement createAnnouncement(Announcement announcement, String username, boolean system) throws IllegalAccessException,
```
You can define another DTO (ActivityAnnouncement) that inherits from `Announcement` that defines templateParams as a new attribute.